### ISSUE TITLE
Mesh.ConnectTimeout overwrites Mesh.DefaultConfig.ConnectTimeout

### ISSUE THREAD

**Issue by vvraskin** [istio/istio#11319]
**Describe the bug**
When defining `Mesh.DefaultConfig.ConnectTimeout` it is getting overwritten by `Mesh.ConnectTimeout's` default. 

**Expected behavior**
The property specified by the user is not getting overwritten, e.g. the `Mesh.DefaultConfig.ConnectionTimeout` is used

**Steps to reproduce the bug**
1. Deploy latest istio version with `Mesh.DefaultConfig.ConnectionTimeout` set to 10s
2. Exec to one of the envoy containers
3. Run `curl localhost:15000/config-dump`
4. Observe `connect_timeout` for dynamic clusters being set to 1s (the default Mesh.ConnectionTimeout)

**Version**
Kube 1.11.5, istio 1.0.2 (the latest version was verified as well)

**Installation**
kubectl apply [istio.yaml](https://github.com/knative/serving/blob/master/third_party/istio-1.0.2/istio.yaml)

**Environment**
IBM Cloud

**Comment by GregHanson** [comment]
a little more details on this:
timeout is configured to 10s here in mesh config https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/templates/configmap.yaml#L91
If not provided in the mesh config, 1s timeout is used https://github.com/istio/istio/blob/master/pilot/pkg/model/context.go#L357
but here and here, the default cluster for services are set to 1s regardless of mesh config/default setting:
https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/cluster.go#L1054
https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/cluster.go#L1066

Is the connect_timeout field only for TCP services?   That would explain why in a config dump of the bookinfo sample, 10s appears to only be applied to GRPC clusters for istio-pilot:
```
{
     "cluster": {
      "name": "xds-grpc",
      "type": "STRICT_DNS",
      "connect_timeout": "10s",
      "hosts": [
       {
        "socket_address": {
         "address": "istio-pilot.istio-system",
         "port_value": 15010
        }
       }
      ],
```
Is 1s the default connect timeout for HTTP services? @rshriram @costinm @frankbu

**Comment by ayj** [comment]
cc @andraxylia @costinm

**Comment by linsun** [comment]
think this can be closed now due to #12089, pls reopen if needed
